omnigibson/prims/entity_prim.py

```diff
@@ -954,76 +954,138 @@ def get_relative_linear_velocity(self):
+            else:
+                if position is not None:
+                    position = np.asarray(position)[None, :]
+                if orientation is not None:
+                    orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
+                self._articulation_view.set_world_poses(position, orientation)
+                PoseAPI.invalidate()
+
+        elif frame == RelativeFrame.SCENE:
+
+            # TODO: Implement this for scene frame
+            pass
+
+        elif frame == RelativeFrame.PARENT:
+
+            # If kinematic only, clear cache for the root link
+            if self.kinematic_only:
+                self.root_link.clear_kinematic_only_cache()
+            # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
+            if og.sim.is_stopped():
+                return XFormPrim.set_position_orientation(self, position, orientation, frame=frame)
+            # Delegate to RigidPrim if we are not articulated
+            elif self._articulation_view is None:
+                self.root_link.set_position_orientation(position=position, orientation=orientation, frame=frame)
+            # Sim is running and articulation view exists, so use that physx API backend
+            else:
+                if position is not None:
+                    position = np.asarray(position)[None, :]
+                if orientation is not None:
+                    orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
+                self._articulation_view.set_local_poses(position, orientation)
+                PoseAPI.invalidate()
 
-    def get_position_orientation(self):
-        # If the simulation isn't running, we should read from this prim's XForm (object-level) properties directly
-        if og.sim.is_stopped():
-            return XFormPrim.get_position_orientation(self)
-        # Delegate to RigidPrim if we are not articulated
-        elif self._articulation_view is None:
-            return self.root_link.get_position_orientation()
-        # Sim is running and articulation view exists, so use that physx API backend
         else:
-            positions, orientations = self._articulation_view.get_world_poses()
-            return positions[0], orientations[0][[1, 2, 3, 0]]
+            raise ValueError(f"Invalid frame: {frame}")
+
+    def get_position_orientation(self, frame=RelativeFrame.WORLD):
+        """
+        Gets prim's pose with respect to the specified frame.
+
+        Args:
+            frame (RelativeFrame): frame to get the pose with respect to. Default to WORLD. PARENT frame
+            get position relative to the object parent. SCENE frame get position relative to the scene.
+
+        Returns:
+            2-tuple:
+                - 3-array: (x,y,z) position in the specified frame
+                - 4-array: (x,y,z,w) quaternion orientation in the specified frame
+        """
+
+        if frame == RelativeFrame.WORLD:
```

Comment:
Here too doing the branching up high doesn't make sense. The first two options inside this branch are repeated verbatim in the last branch. Don't repeat code. Branch in just the articulation_view-getting branch.